### PR TITLE
Convert list tables in migration guides to Markdown tables

### PR DESCRIPTION
**Pull request by Eric-Arellano**
Part of https://github.com/Qiskit/documentation/issues/213.

I used the below script. Warning: it doesn't handle when the list item has multiple lines for the entry. Those were manually fixed in the second commit.

```python
from __future__ import annotations

from dataclasses import dataclass
from pathlib import Path
from typing import Iterator


GUIDES = [
    Path(f"docs/api/migration-guides/qiskit-{guide}.mdx")
    for guide in ("algorithms-module", "opflow-module", "quantum-instance")
]


@dataclass
class Table:
    header: list[str]
    entries: list[list[str]]

    @classmethod
    def from_rst(cls, rst: str) -> Table:
        parsed_rows: list[list[str]] = []
        current_row: list[str] = []

        for line in rst.splitlines():
            line = line.strip()
            if line.startswith("* -"):  # new row
                # Before processing the new row, save the previous row.
                if current_row:
                    parsed_rows.append(current_row)
                current_row = [line.replace("* -", "").strip()]
            elif line.startswith("-"):  # continue current_row
                current_row.append(line.replace("-", "").strip())

        # Save the final row (ince normally a row only
        # gets saved once we already have started parsing
        # a new row).
        if current_row:
            parsed_rows.append(current_row)

        return cls(header=parsed_rows[0], entries=parsed_rows[1:])

    def to_md(self) -> str:
        result = "| " + " | ".join(self.header) + " |\n"
        result += "| " + " | ".join(["---"] * len(self.header)) + " |\n"
        for entry in self.entries:
            result += "| " + " | ".join(entry) + " |\n"
        
        # Convert RST to MD
        result = result.replace("``", "`")
        return result


def parse_rst_table(lines: list[str]) -> str:
    result = []
    while lines and not lines[0].startswith("```"):
        result.append(lines.pop(0))
    # Remove the final ``` line
    lines.pop(0)
    return "\n".join(result)


def process_lines(lines: list[str]) -> Iterator[str]:
    """Replace RST tables with Markdown.

    This function intentionally uses `pop()` to iterate through every line. If
    we encounter a list-table, we use parse_rst_table() to replace the entire
    block with the markdown table. Otherwise, we simply preserve the line.
    """
    while lines:
        line = lines.pop(0)
        if line.startswith("```{eval-rst}") and lines[0].startswith(".. list-table::"):
            rst_block = parse_rst_table(lines)
            converted = Table.from_rst(rst_block).to_md()
            yield from converted.splitlines()
        else:
            yield line


def main() -> None:
    for guide in GUIDES:
        original = guide.read_text().splitlines()
        updated = list(process_lines(original))
        guide.write_text("\n".join(updated) + "\n")


main()
```

### CODE REVIEW

**Comment by Eric-Arellano** [comment]
To review, you can preview:

- https://qiskit-docs-preview-pr-296.1799mxdls7qz.us-south.codeengine.appdomain.cloud/api/migration-guides/qiskit-opflow-module
- https://qiskit-docs-preview-pr-296.1799mxdls7qz.us-south.codeengine.appdomain.cloud/api/migration-guides/qiskit-quantum-instance

File: docs/api/migration-guides/qiskit-opflow-module.mdx [hunk 1]
```diff
@@ -455,38 +393,15 @@ to initialize it.
...
-     - This class was used to combine a [`qiskit.opflow.primitive_ops.PauliSumOp`](../qiskit/qiskit.opflow.primitive_ops.PauliSumOp) with its identified symmetries in one object.
-       This functionality is not currently used in any workflow, and has been deprecated without replacement.
-       See [`qiskit.quantum_info.analysis.Z2Symmetries`](../qiskit/qiskit.quantum_info.Z2Symmetries) example for updated workflow.
-
-   * - [`qiskit.opflow.primitive_ops.Z2Symmetries`](../qiskit/qiskit.opflow.primitive_ops.Z2Symmetries)
-     - [`qiskit.quantum_info.analysis.Z2Symmetries`](../qiskit/qiskit.quantum_info.Z2Symmetries). See example :ref:`below <example_z2_sym>`.
-```
+| Opflow | Alternative |
+| --- | --- |
+| [`qiskit.opflow.primitive_ops.PrimitiveOp`](../qiskit/qiskit.opflow.primitive_ops.PrimitiveOp) | As mentioned above, this class is used to generate an instance of one of the classes below, so there is |
```
Comment (frankharkins) [inline docs/api/migration-guides/qiskit-opflow-module.mdx:398]:
```suggestion
| [`qiskit.opflow.primitive_ops.PrimitiveOp`](../qiskit/qiskit.opflow.primitive_ops.PrimitiveOp) | As mentioned above, this class is used to generate an instance of one of the classes below, so there is no direct replacement |
```

File: docs/api/migration-guides/qiskit-opflow-module.mdx [hunk 2]
```diff
@@ -633,29 +548,12 @@ The [`qiskit.opflow.list_ops`](../qiskit/qiskit.opflow.list_ops) module containe
...
-   * - [`qiskit.opflow.list_ops.SummedOp`](../qiskit/qiskit.opflow.list_ops.SummedOp)
-     - No direct replacement. For [`qiskit.quantum_info.Pauli`](../qiskit/qiskit.quantum_info.Pauli) operators, use [`qiskit.quantum_info.SparsePauliOp`](../qiskit/qiskit.quantum_info.SparsePauliOp).
-
-   * - [`qiskit.opflow.list_ops.TensoredOp`](../qiskit/qiskit.opflow.list_ops.TensoredOp)
-     - No direct replacement. For [`qiskit.quantum_info.Pauli`](../qiskit/qiskit.quantum_info.Pauli) operators, use [`qiskit.quantum_info.SparsePauliOp`](../qiskit/qiskit.quantum_info.SparsePauliOp).
-
-```
+| Opflow | Alternative |
+| --- | --- |
+| [`qiskit.opflow.list_ops.ListOp`](../qiskit/qiskit.opflow.list_ops.ListOp) | No direct replacement. This is the base class for operator lists. In general, these could be replaced with |
```
Comment (frankharkins) [inline docs/api/migration-guides/qiskit-opflow-module.mdx:553]:
```suggestion
| [`qiskit.opflow.list_ops.ListOp`](../qiskit/qiskit.opflow.list_ops.ListOp) | No direct replacement. This is the base class for operator lists. In general, these could be replaced with Python ``list``\s. For [`qiskit.quantum_info.Pauli`](../qiskit/qiskit.quantum_info.Pauli) operators, there are a few alternatives, depending on the use-case. |
```

**Review by frankharkins — CHANGES_REQUESTED** [review]
It looks like this doesn't recognise cells broken over more than one line, e.g.

```
   * - Heading
     - This text should all appear
       in the same table cell
```

I've pointed out a couple of examples.

**Review by kevinsung — APPROVED** [review]

**Pull request merged**